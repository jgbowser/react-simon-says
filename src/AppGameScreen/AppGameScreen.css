.game_screen_app {
  height: 100%;
}