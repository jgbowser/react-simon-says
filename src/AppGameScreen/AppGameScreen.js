import React from 'react'
import GameBoard from '../GameBoard/GameBoard'
import RoundHeader from '../RoundHeader/RoundHeader'
import TurnIndicator from '../TurnIndicator/TurnIndicator'
import GameContext from '../GameContext'
import './AppGameScreen.css'

export default class AppGameScreen extends React.Component {
  static contextType = GameContext
  render() {
    const { turn } = this.context
    return (
      <div className="game_screen_app">
        <RoundHeader />
        <GameBoard />
        <TurnIndicator player={turn} />
        {/* conditionally rendered fail screen with link to end screen when player loses */}
      </div>
    )
  }
}