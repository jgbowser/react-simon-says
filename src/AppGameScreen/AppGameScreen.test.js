import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import AppGameScreen from './AppGameScreen'

describe('AppGameScreen Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<AppGameScreen />, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<AppGameScreen />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})