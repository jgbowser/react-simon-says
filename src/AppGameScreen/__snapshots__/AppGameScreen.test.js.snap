// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppGameScreen Component renders the UI as expected 1`] = `
<div
  className="game_screen_app"
>
  <header
    className="round_header"
  >
    <h1>
      Round: 
      
    </h1>
  </header>
  <div
    className="gameboard_container"
  >
    <div
      className="top_row_pads"
    >
      <div
        className="game_pad"
        id="1"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "green",
          }
        }
      />
      <div
        className="game_pad"
        id="2"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
    </div>
    <div
      className="bottom_row_pads"
    >
      <div
        className="game_pad"
        id="3"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "yellow",
          }
        }
      />
      <div
        className="game_pad"
        id="4"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "blue",
          }
        }
      />
    </div>
  </div>
  <div
    className="turn_indicator"
  >
    <p>
      Simon's turn. Pay attention!
    </p>
  </div>
</div>
`;
